streams/src/test/java/org/apache/kafka/streams/kstream/internals/SuppressScenarioTest.java

```diff
@@ -225,36 +221,15 @@ public void shouldSuppressIntermediateEventsWithTimeLimit() {
                     new KeyValueTimestamp<>("tick", 1L, 4L)
                 )
             );
+            // tick is still buffered, since it was first inserted at time 3, and it is only time 4 right now.
             verify(
                 drainProducerRecords(driver, "output-suppressed", STRING_DESERIALIZER, LONG_DESERIALIZER),
-                singletonList(
```

Comment:
What was the original intend of this part? And why don't we need it?

Reply:
When we enforced constraints before buffering, we needed one extra tick to flush everything out. Now that we buffer first, everything happens more promptly, so we don't need this last cycle to witness all the results we're looking for.